netci.groovy

```diff
@@ -658,35 +658,52 @@ branchList.each { branchName ->
     }
 }
 
-// Generate the build and test versions for Windows_NT.  When full build/run is supported on a platform, those platforms
+// Generate the build and test versions for Windows_NT and Ubuntu.  When full build/run is supported on a platform, those platforms
 // could be removed from above and then added in below.
-def supportedFullCyclePlatforms = ['Windows_NT']
+def supportedFullCyclePlatforms = ['Windows_NT', 'Ubuntu']
 
 branchList.each { branchName ->
     ['Debug', 'Release'].each { configurationGroup ->
-        supportedFullCyclePlatforms.each { osGroup ->
+        supportedFullCyclePlatforms.each { os ->
+            def osGroup = osGroupMap[os]
             def isPR = (branchName == 'pr') 
-            def newJobName = "${osGroup.toLowerCase()}_${configurationGroup.toLowerCase()}"
+            def newJobName = "${os.toLowerCase()}_${configurationGroup.toLowerCase()}"
 
             def newJob = job(getJobName(Utilities.getFullJobName(project, newJobName, isPR), branchName)) {
                 steps {
-                    batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd /p:ConfigurationGroup=${configurationGroup} /p:OSGroup=${osGroup}")
-                    batchFile("C:\\Packer\\Packer.exe .\\bin\\build.pack .\\bin")
+                    if (os == 'Windows 10' || os == 'Windows 7' || os == 'Windows_NT') {
+                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd /p:ConfigurationGroup=${configurationGroup} /p:OSGroup=${osGroup}")
+                        batchFile("C:\\Packer\\Packer.exe .\\bin\\build.pack .\\bin")
+                    }
+                    else if (os == 'OSX') {
+                        shell("HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:TestWithLocalLibraries=true")
+                    }
+                    else {
+                        shell("sudo HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:TestWithLocalLibraries=true")
+                    }
                 }
             }
 
-            // Set the affinity.  All of these run on Windows currently.
-            Utilities.setMachineAffinity(newJob, osGroup, 'latest-or-auto')
+            // Set the affinity.
+            Utilities.setMachineAffinity(newJob, os, 'latest-or-auto')
             // Set up standard options.
             Utilities.standardJobSetup(newJob, project, isPR, getFullBranchName(branchName))
             // Add the unit test results
             Utilities.addXUnitDotNETResults(newJob, 'bin/tests/**/testResults.xml')
+            def archiveContents = ''
+            if (os == 'Windows 10' || os == 'Windows 7' || os == 'Windows_NT') {
+                archiveContents = "bin/build.pack,bin/${osGroup}.AnyCPU.Debug/**,bin/ref/**,bin/packages/**,msbuild.log"
```

Comment:
Yeah, @ellismg noticed this too. I think we can fix this separately, since I'm just matching what Windows is doing here. We should fix it in all places it occurs in the file.